### PR TITLE
[clone] Split runtime into devnet and mainnet

### PR DESCRIPTION
**Pull request by weezy20**
This PR is a clone of #30 where we have rebased the changes on top of current `main:HEAD` Since this is a huge rebase and the [other PR](#30) has been built without the many changes interim, it's best to keep this one separate, for nice and clean observation.

Issue #10

### CODE REVIEW

**Comment by evilrobot-01** [comment]
Minor thing, but do we need the word template in the runtime names? It would be easier for downstream teams if it were omitted and it's implied based on the repo name.

Then template users don't need to do anything to remove the template name when adopting, making it easier.

A counter argument is that a template perhaps provides no guarantees, so think it needs to be carefully considered.

**Comment by weezy20** [comment]
> Minor thing, but do we need the word template in the runtime names? It would be easier for downstream teams if it were omitted and it's implied based on the repo name.

Template is supposed to be a placeholder. My take on this would be, given that it's a `template` we are free to choose whatever name we use for spec, impl strs and the runtime. Renaming things for a team would be as simple as replacing the entire string in the directory so I don't see renaming it bringing much convenience.

File: runtime/common/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+# [build-dependencies]
+# substrate-wasm-builder = {git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.42"}
+
+[dependencies]
+codec = {package = "parity-scale-codec", version = "3.0.0", default-features = false, features = ["derive"]}
+scale-info = {version = "2.5.0", default-features = false, features = ["derive"]}
+
+# Substrate
+frame-support = {git = "https://github.com/paritytech/substrate", default-features = false, branch = "polkadot-v0.9.42"}
+# frame-system = {git = "https://github.com/paritytech/substrate", default-features = false, branch = "polkadot-v0.9.42"}
```
Comment (stiiifff) [inline runtime/common/Cargo.toml:22]: Remove commented out deps if not needed.

File: runtime/devnet/Cargo.toml [hunk 2]
```diff
@@ -1,12 +1,12 @@
 [package]
+name = "devnet-template-runtime"
```
Comment (stiiifff) [inline runtime/devnet/Cargo.toml:2]: As mentioned by @evilrobot-01, I would remove the `template` term.

File: runtime/common/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+[package]
+authors = ["Anonymous"]
+description = "Template Runtime Common"
+edition = "2021"
+license = "Unlicense"
+name = "template-common"
```
Comment (stiiifff) [inline runtime/common/Cargo.toml:6]: `runtime-common` would be a better name.

**Review by stiiifff — CHANGES_REQUESTED** [review]

File: node/src/command.rs [hunk 4]
```diff
@@ -17,15 +17,74 @@ use sp_runtime::traits::{AccountIdConversion, Block as BlockT};
...
+}
+
 fn load_spec(id: &str) -> std::result::Result<Box<dyn ChainSpec>, String> {
 	Ok(match id {
-		"dev" => Box::new(chain_spec::development_config()),
-		"template-rococo" => Box::new(chain_spec::local_testnet_config()),
-		"" | "local" => Box::new(chain_spec::local_testnet_config()),
-		path => Box::new(chain_spec::ChainSpec::from_json_file(std::path::PathBuf::from(path))?),
+		"dev" => Box::new(chain_spec::devnet::development_config()),
+		"template-rococo" => Box::new(chain_spec::devnet::local_testnet_config()),
```
Comment (stiiifff) [inline node/src/command.rs:75]: Shouldn't this be called `devnet-local` ?

File: runtime/devnet/src/lib.rs [hunk 5]
```diff
@@ -210,8 +202,8 @@ impl_opaque_keys! {
 
 #[sp_version::runtime_version]
 pub const VERSION: RuntimeVersion = RuntimeVersion {
-	spec_name: create_runtime_str!("template-parachain"),
-	impl_name: create_runtime_str!("template-parachain"),
+	spec_name: create_runtime_str!("template-devnet"),
+	impl_name: create_runtime_str!("template-devnet"),
```
Comment (stiiifff) [inline runtime/devnet/src/lib.rs:206]: Remove `template` prefix.

**Review by stiiifff — CHANGES_REQUESTED** [review]

File: node/src/command.rs (hunk 4 above)
Comment (weezy20) [reply node/src/command.rs:75]: Yes it should. Adding `devnet-local` here.

**Review by stiiifff — APPROVED** [review]

**Comment by stiiifff** [comment]
LGTM !

**Pull request merged**